src/wallet/rpcwallet.cpp

```diff
@@ -887,12 +897,28 @@ UniValue sc_create(const UniValue& params, bool fHelp)
         }
     }
 
-    if (params.size() > 6)
+    if (params.size() > 7)
     {
-        const std::string& inputString = params[6].get_str();
+        uint8_t cswProvingSystemIndex = params[7].get_int();
+
+        if (cswProvingSystemIndex < static_cast<uint8_t>(Sidechain::ProvingSystemType::CoboundaryMarlin) ||
```

Comment:
Consider using an exact match for all supported enum values instead of a range check. The same applies for analogous checks in this file.